enterprise/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/PageCacheWarmer.java

```diff
@@ -298,61 +307,17 @@ private ExecutorService buildExecutorService( JobScheduler scheduler )
                 threadFactory, rejectionPolicy );
     }
 
-    private InputStream compressedInputStream( File input ) throws IOException
+    private Stream<Profile> filterRelevant( Profile[] profiles, PagedFile pagedFile )
     {
-        InputStream source = fs.openAsInputStream( input );
-        try
-        {
-            return new GZIPInputStream( source );
-        }
-        catch ( IOException e )
-        {
-            IOUtils.closeAllSilently( source );
-            throw new IOException( "Exception when building decompressor.", e );
-        }
-    }
-
-    private OutputStream compressedOutputStream( File output ) throws IOException
-    {
-        fs.mkdirs( output.getParentFile() );
-        StoreChannel channel = fs.open( output, OpenMode.READ_WRITE );
-        ByteBuffer buf = ByteBuffer.allocate( 1 );
-        OutputStream sink = new OutputStream()
-        {
-            @Override
-            public void write( int b ) throws IOException
-            {
-                buf.put( (byte) b );
-                buf.flip();
-                channel.write( buf );
-                buf.flip();
-            }
-
-            @Override
-            public void close() throws IOException
-            {
-                channel.truncate( channel.position() );
-                channel.close();
-            }
-        };
-        try
-        {
-            return new GZIPOutputStream( sink );
-        }
-        catch ( IOException e )
-        {
-            // We close the channel instead of the sink here, because we don't want to truncate the file if we fail
-            // to open the gzip output stream.
-            IOUtils.closeAllSilently( channel );
-            throw new IOException( "Exception when building compressor.", e );
-        }
+        return Stream.of( profiles ).filter( Profile.relevantTo( pagedFile ) );
     }
 
-    private File profileOutputFileName( PagedFile file )
+    private Profile[] findExistingProfiles( List<PagedFile> pagedFiles )
```

Comment:
Now when we have a separate folder we could have something like `profiles/cacheprof/*` and just list all the files in that folder.

Reply:
We're only interested in profiles for files that are mapped. We don't _know_ if there will always only be one "profiles" directory, since it is placed as a sibling to the mapped files, which can in principle be in any number of directories. And if people place other random files in the "profiles" directory, I'd like to ignore them.